movebucket_test.go

```diff
@@ -216,6 +216,151 @@ func TestTx_MoveBucket(t *testing.T) {
+	db2 := btesting.MustCreateDBWithOption(t, &bbolt.Options{PageSize: pageSize})
+	err = db2.Update(func(tx *bbolt.Tx) error {
+		prepareBuckets(t, tx, dstBucketPath...)
+		return nil
+	})
+	db2.MustClose()
+	require.NoError(t, err)
+
+	t.Log("Re-opening source bucket from DB1")
+	srcDB, sErr := bbolt.Open(db1.Path(), 0600, bbolt.DefaultOptions)
+	require.NoError(t, sErr)
+	defer func() {
+		cErr := srcDB.Close()
+		require.NoError(t, cErr)
+	}()
+
+	t.Log("Reading source bucket in a separate RWTx")
+	sTx, sErr := srcDB.Begin(true)
+	require.NoError(t, sErr)
+	defer func() {
+		rErr := sTx.Rollback()
+		require.NoError(t, rErr)
+	}()
+	srcBucket = prepareBuckets(t, sTx, srcBucketPath...)
+
+	t.Log("Re-opening target bucket from DB2")
+	dstDB, dErr := bbolt.Open(db2.Path(), 0600, bbolt.DefaultOptions)
+	require.NoError(t, dErr)
+	defer func() {
+		cErr := dstDB.Close()
+		require.NoError(t, cErr)
+	}()
+
+	t.Log("Moving the sub-bucket in a separate RWTx")
+	err = dstDB.Update(func(tx *bbolt.Tx) error {
+		dstBucket := prepareBuckets(t, tx, dstBucketPath...)
+		mErr := srcBucket.MoveBucket([]byte(bucketToMove), dstBucket)
+		require.Equal(t, expectedErr, mErr)
+
+		return nil
+	})
+	require.NoError(t, err)
+}
+
+func TestBucket_MoveBucket_DiffTx(t *testing.T) {
+	testCases := []struct {
+		name          string
+		srcBucketPath []string
+		srcRWTx       bool
+		bucketToMove  string
+		dstBucketPath []string
+		dstRWTx       bool
+		expectedErr   error
+	}{
+		{
+			name:          "src is RWTx and target is RTx",
+			srcRWTx:       true,
+			srcBucketPath: []string{"sb1", "sb2"},
+			bucketToMove:  "bucketToMove",
+			dstBucketPath: []string{"db1", "db2"},
+			dstRWTx:       false,
+			expectedErr:   errors.ErrTxNotWritable,
+		},
+		{
+			name:          "src is RTx and target is RWTx",
+			srcRWTx:       false,
+			srcBucketPath: []string{"sb1", "sb2"},
+			bucketToMove:  "bucketToMove",
+			dstBucketPath: []string{"db1", "db2"},
+			dstRWTx:       true,
+			expectedErr:   errors.ErrTxNotWritable,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var srcBucket *bbolt.Bucket
+			var dstBucket *bbolt.Bucket
+
+			t.Log("Creating source and target buckets and populate some data")
+			db := btesting.MustCreateDBWithOption(t, &bbolt.Options{PageSize: pageSize})
+			err := db.Update(func(tx *bbolt.Tx) error {
+				srcBucket = prepareBuckets(t, tx, tc.srcBucketPath...)
+				dstBucket = prepareBuckets(t, tx, tc.dstBucketPath...)
+				return nil
+			})
+			db.MustClose()
+			require.NoError(t, err)
+
+			t.Log("Re-opening the database")
+			dB, dErr := bbolt.Open(db.Path(), 0600, bbolt.DefaultOptions)
+			require.NoError(t, dErr)
+			defer func() {
+				cErr := dB.Close()
+				require.NoError(t, cErr)
+			}()
+
+			t.Log("Re-opening source bucket in a separate Tx")
+			sTx, sErr := dB.Begin(tc.srcRWTx)
+			require.NoError(t, sErr)
+			defer func() {
+				rErr := sTx.Rollback()
```

Comment:
require.NoError(t, sTx.Rollback()) so that we don't need to introduce new var name